C++ unit tests fail to link with subset of test files

Steps to reproduce:
Update `unit_test_srcs` in `CMakeLists.txt` to:
```
    set(unit_test_srcs
            async/test/test_async.cpp
            processing/test/test_component_manager.cpp
            util/test/gtest_main.cpp
    )
```
Build will succeed. Update to:
```
    set(unit_test_srcs
            processing/test/test_component_manager.cpp
            util/test/gtest_main.cpp
    )
```
Linking will fail. Through binary search, identified that [this commit](https://github.com/man-group/ArcticDB/commit/066512c0de91ebc93062b75eb0d8ce06b87a4262) is what breaks things. Linker error:
```
[ 98%] Linking CXX executable test_unit_arcticdb
/usr/bin/ld: libarcticdb_core_static.a(nfs_backed_storage.cpp.o): in function `arcticdb::storage::s3::RealS3Client::RealS3Client<Aws::Auth::AWSCredentials, Aws::Client::ClientConfiguration, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy, bool>(Aws::Auth::AWSCredentials&&, Aws::Client::ClientConfiguration&&, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy&&, bool&&)':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.hpp:32: undefined reference to `Aws::S3::S3Client::S3Client(Aws::Auth::AWSCredentials const&, Aws::Client::ClientConfiguration const&, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy, bool, Aws::S3::US_EAST_1_REGIONAL_ENDPOINT_OPTION)'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `arcticdb::storage::s3::RealS3Client::head_object(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:46: undefined reference to `Aws::S3::Model::HeadObjectRequest::HeadObjectRequest()'
/usr/bin/ld: /home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:48: undefined reference to `Aws::S3::S3Client::HeadObject(Aws::S3::Model::HeadObjectRequest const&) const'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `arcticdb::storage::s3::RealS3Client::get_object(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:120: undefined reference to `Aws::S3::Model::GetObjectRequest::GetObjectRequest()'
/usr/bin/ld: /home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:123: undefined reference to `Aws::S3::S3Client::GetObject(Aws::S3::Model::GetObjectRequest const&) const'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `arcticdb::storage::s3::RealS3Client::put_object(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, arcticdb::Segment&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:140: undefined reference to `Aws::S3::Model::PutObjectRequest::PutObjectRequest()'
/usr/bin/ld: /home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:161: undefined reference to `Aws::S3::S3Client::PutObject(Aws::S3::Model::PutObjectRequest const&) const'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `arcticdb::storage::s3::RealS3Client::delete_objects(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:176: undefined reference to `Aws::S3::Model::DeleteObjectsRequest::DeleteObjectsRequest()'
/usr/bin/ld: /home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:178: undefined reference to `Aws::S3::Model::Delete::Delete()'
/usr/bin/ld: /home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:181: undefined reference to `Aws::S3::Model::ObjectIdentifier::ObjectIdentifier()'
/usr/bin/ld: /home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:187: undefined reference to `Aws::S3::S3Client::DeleteObjects(Aws::S3::Model::DeleteObjectsRequest const&) const'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `arcticdb::storage::s3::RealS3Client::list_objects(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >) const':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:211: undefined reference to `Aws::S3::Model::ListObjectsV2Request::ListObjectsV2Request()'
/usr/bin/ld: /home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.cpp:217: undefined reference to `Aws::S3::S3Client::ListObjectsV2(Aws::S3::Model::ListObjectsV2Request const&) const'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `Aws::AmazonWebServiceRequest::~AmazonWebServiceRequest()':
/home/aowens/source/ArcticDB/cpp/out/linux-debug-build/vcpkg_installed/x64-linux/include/aws/core/AmazonWebServiceRequest.h:46: undefined reference to `vtable for Aws::AmazonWebServiceRequest'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `Aws::AmazonSerializableWebServiceRequest::~AmazonSerializableWebServiceRequest()':
/home/aowens/source/ArcticDB/cpp/out/linux-debug-build/vcpkg_installed/x64-linux/include/aws/core/AmazonSerializableWebServiceRequest.h:42: undefined reference to `vtable for Aws::AmazonSerializableWebServiceRequest'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `arcticdb::storage::s3::RealS3Client::~RealS3Client()':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.hpp:29: undefined reference to `Aws::S3::S3Client::~S3Client()'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o):(.data.rel.ro._ZTVN3Aws2S39S3RequestE[_ZTVN3Aws2S39S3RequestE]+0x20): undefined reference to `Aws::AmazonSerializableWebServiceRequest::GetBody() const'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o):(.data.rel.ro._ZTVN3Aws2S39S3RequestE[_ZTVN3Aws2S39S3RequestE]+0x30): undefined reference to `Aws::AmazonWebServiceRequest::GetAdditionalCustomHeaders[abi:cxx11]() const'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o):(.data.rel.ro._ZTVN3Aws2S39S3RequestE[_ZTVN3Aws2S39S3RequestE]+0x38): undefined reference to `Aws::AmazonWebServiceRequest::SetAdditionalCustomHeaderValue(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o):(.data.rel.ro._ZTVN3Aws2S39S3RequestE[_ZTVN3Aws2S39S3RequestE]+0x110): undefined reference to `Aws::AmazonWebServiceRequest::GetEndpointContextParams() const'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o):(.data.rel.ro._ZTIN3Aws2S39S3RequestE[_ZTIN3Aws2S39S3RequestE]+0x10): undefined reference to `typeinfo for Aws::AmazonSerializableWebServiceRequest'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `Aws::S3::Model::HeadObjectRequest::~HeadObjectRequest()':
/home/aowens/source/ArcticDB/cpp/out/linux-debug-build/vcpkg_installed/x64-linux/include/aws/s3/model/HeadObjectRequest.h:29: undefined reference to `vtable for Aws::S3::Model::HeadObjectRequest'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `Aws::S3::Model::GetObjectRequest::~GetObjectRequest()':
/home/aowens/source/ArcticDB/cpp/out/linux-debug-build/vcpkg_installed/x64-linux/include/aws/s3/model/GetObjectRequest.h:29: undefined reference to `vtable for Aws::S3::Model::GetObjectRequest'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `Aws::S3::Model::PutObjectRequest::~PutObjectRequest()':
/home/aowens/source/ArcticDB/cpp/out/linux-debug-build/vcpkg_installed/x64-linux/include/aws/s3/model/PutObjectRequest.h:35: undefined reference to `vtable for Aws::S3::Model::PutObjectRequest'
/usr/bin/ld: /home/aowens/source/ArcticDB/cpp/out/linux-debug-build/vcpkg_installed/x64-linux/include/aws/s3/model/PutObjectRequest.h:35: undefined reference to `Aws::AmazonStreamingWebServiceRequest::~AmazonStreamingWebServiceRequest()'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `Aws::S3::Model::DeleteObjectsRequest::~DeleteObjectsRequest()':
/home/aowens/source/ArcticDB/cpp/out/linux-debug-build/vcpkg_installed/x64-linux/include/aws/s3/model/DeleteObjectsRequest.h:29: undefined reference to `vtable for Aws::S3::Model::DeleteObjectsRequest'
/usr/bin/ld: libarcticdb_core_static.a(s3_real_client.cpp.o): in function `Aws::S3::Model::ListObjectsV2Request::~ListObjectsV2Request()':
/home/aowens/source/ArcticDB/cpp/out/linux-debug-build/vcpkg_installed/x64-linux/include/aws/s3/model/ListObjectsV2Request.h:30: undefined reference to `vtable for Aws::S3::Model::ListObjectsV2Request'
/usr/bin/ld: libarcticdb_core_static.a(s3_storage.cpp.o): in function `arcticdb::storage::s3::RealS3Client::RealS3Client<Aws::Client::ClientConfiguration, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy, bool>(Aws::Client::ClientConfiguration&&, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy&&, bool&&)':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.hpp:32: undefined reference to `Aws::S3::S3Client::S3Client(Aws::Client::ClientConfiguration const&, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy, bool, Aws::S3::US_EAST_1_REGIONAL_ENDPOINT_OPTION)'
/usr/bin/ld: libarcticdb_core_static.a(s3_storage.cpp.o): in function `arcticdb::storage::s3::RealS3Client::RealS3Client<Aws::Auth::AWSCredentials&, Aws::Client::ClientConfiguration, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy, bool>(Aws::Auth::AWSCredentials&, Aws::Client::ClientConfiguration&&, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy&&, bool&&)':
/home/aowens/source/ArcticDB/cpp/arcticdb/storage/s3/s3_real_client.hpp:32: undefined reference to `Aws::S3::S3Client::S3Client(Aws::Auth::AWSCredentials const&, Aws::Client::ClientConfiguration const&, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy, bool, Aws::S3::US_EAST_1_REGIONAL_ENDPOINT_OPTION)'
collect2: error: ld returned 1 exit status
make[3]: *** [arcticdb/CMakeFiles/test_unit_arcticdb.dir/build.make:186: arcticdb/test_unit_arcticdb] Error 1
make[2]: *** [CMakeFiles/Makefile2:1284: arcticdb/CMakeFiles/test_unit_arcticdb.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1291: arcticdb/CMakeFiles/test_unit_arcticdb.dir/rule] Error 2
make: *** [Makefile:660: test_unit_arcticdb] Error 2
```

--- a/cpp/arcticdb/CMakeLists.txt
+++ b/cpp/arcticdb/CMakeLists.txt
@@ -896,7 +896,6 @@ if(${TEST})
             ${COMMON_PUBLIC_TEST_LIBRARIES}
             PRIVATE
             ${AWSSDK_LINK_LIBRARIES}
-            arcticdb_core_static
             )
 
         target_include_directories(test_unit_arcticdb PRIVATE

